lark/load_grammar.py

```diff
@@ -328,9 +328,9 @@ def pattern(self, p):
                 try:
                     term_name = _TERMINAL_NAMES[value]
                 except KeyError:
-                    if value.isalnum() and value[0].isalpha() and value.upper() not in self.term_set:
+                    if isalnum(value) and isalpha(value[0]) and value.upper() not in self.term_set:
                         with suppress(UnicodeEncodeError):
-                            value.upper().encode('ascii') # Make sure we don't have unicode in our terminal names
+                            value.upper().encode('utf8')  # Why shouldn't we have unicode in our terminal names?
```

Comment:
What is the point of this line? And especially the comment?

Reply:
I had just been trying to make minimal changes, but I think you're right in that this line doesn't do anything anymore. Tests all pass without it so I removed these few lines. Thank you!